### ISSUE TITLE
AmazonSQS Transport usage sample implies AWS_REGION environment var can be used to specify AWS region

### ISSUE THREAD

**Issue by mauroservienti** [Particular/NServiceBus.AmazonSQS#186]
Related to AmazonSQS Transport V3.x

In [Simple AmazonSQS Transport usage](https://docs.particular.net/samples/sqs/simple/) we state:

> Add the AWS Access Key ID and AWS Secret Access Key to the following environment variables:
> * Access Key ID in AWS_ACCESS_KEY_ID
> * Secret Access Key in AWS_SECRET_ACCESS_KEY
> * Default Region in AWS_REGION

However the `AWS_REGION` environment variable is never used by the 3.x version of the transport. At the same time in the [configuration options](https://docs.particular.net/transports/sqs/configuration-options#region) documentation section we state:

> Region
> Mandatory
> Default: None.

that seems to be in conflict with the sample documentation. Looking at the code we effectively never try to read a `AWS_REGION` environment variable and rely on users calling the `Region` configuration option.

We could:

* fix documentation by removing what's stated in the sample and updating it with the `Region` configuration option.
* add support for the `AWS_REGION` environment variable e.g. by patching the following code:

https://github.com/Particular/NServiceBus.AmazonSQS/blob/deff356da78937cdea00dca846b7f5a4f0a6ea46/src/NServiceBus.AmazonSQS/ConnectionConfiguration.cs#L15-L25

doing something like:

```
public RegionEndpoint Region
{
   get
   {
      if (region == null)
      {
         region = settings.Get<RegionEndpoint>(SettingsKeys.Region);
      }

      if (region == null)
      {
         //try read from env var
      }

      return region;
   }
}
```

(can be greatly optimized)

**Comment by mikeminutillo** [comment]
Looks like this should be AWS_DEFAULT_REGION maybe https://docs.aws.amazon.com/cli/latest/userguide/cli-environment.html

The DEV version of the transport doesn't contain `.Region()`. How does it behave?

**Comment by mauroservienti** [comment]
All the magic happens in the AWS Client SDK

**Comment by danielmarbach** [comment]
There is great confusion caused by Amazon. The SDK uses AWS_REGION while
the CLI uses AWS_DEFAULT_REGION. I can dig the relevant sources when back
on Monday

Am 04.01.2018 13:32 schrieb "Mauro Servienti" <notifications@github.com>:

**Comment by mikeminutillo** [comment]
OK. Then I think it makes sense to backport that behavior to this version?

**Comment by danielmarbach** [comment]
Here the relevant doco page

https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/setup-credentials.html

applies also for .NET

Default region is mostly for the CLI but some SDKs support both env variables.

https://github.com/ex-aws/ex_aws/issues/365

Preference if I understood correctly is `AWS_REGION`

**Comment by danielmarbach** [comment]
I don't think it makes sense to backport ENV region support to 3.x since the upgrade guide has all the information in there to show the necessary steps https://docs.particular.net/transports/upgrades/amazonsqs-3to4#region

I will send in a PR to read the `AWS_REGION` environment variable in the 3.x and 2.x sample. This eases the transition to v4.

**Comment by danielmarbach** [comment]
Raised https://github.com/Particular/docs.particular.net/pull/3548
